src/libs/elektra/keyname.c

```diff
@@ -508,25 +557,17 @@ ssize_t keySetName (Key * key, const char * newName)
 
 	// from now on this function CANNOT fail -> we may modify the key
 
-	if (test_bit (key->flags, KEY_FLAG_MMAP_KEY))
-	{
-		// key was in mmap region, clear flag and set NULL to allow realloc
-		key->key = NULL;
-		key->keySize = 0;
-		key->ukey = NULL;
-		key->keyUSize = 0;
-		clear_bit (key->flags, (keyflag_t) KEY_FLAG_MMAP_KEY);
-	}
+	keyDetachKeyName (key);
```

Comment:
The full detach is not necessary here. Like the old code shows, we don't need to copy the existing name. We just need a pointer value that is valid for `realloc`. Setting the pointer to `NULL` fulfils this purpose.

Since `keySetName` is called _a lot_, we really should avoid allocations here as far as possible.